### PR TITLE
Update Ball Placement to let robot move through any motion constraints

### PR DESCRIPTION
**Pull request by itsarune**
### Description

Currently, ball placement robots are bound by motion constraints such as DEFENCE_AREAS and FIELD_BOUNDARY_ZONES, which prevent us from us doing ball placement near these areas. This PR lets ball placement robots violate any motion constraint.

### Testing Done

- More tests in ball_placement_play_test.py, existing pass rate maintained and visually looks okay

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/software/ai/motion_constraint/motion_constraint_visitor.cpp [hunk 1]
```diff
@@ -85,6 +85,15 @@ void MotionConstraintVisitor::visit(const MoveGoalieToGoalLineTactic &tactic)
     });
 }
 
+void MotionConstraintVisitor::visit(const BallPlacementTactic &tactic)
+{
+    current_allowed_constraints = std::set<TbotsProto::MotionConstraint>();   
+    for (int i = TbotsProto::MotionConstraint_MIN; i < TbotsProto::MotionConstraint_MAX; ++i)
+    {
+        current_allowed_constraints.emplace(static_cast<TbotsProto::MotionConstraint>(i));
+    }
```
Comment (nimazareian) [inline src/software/ai/motion_constraint/motion_constraint_visitor.cpp:95]: The only motion constraint that we probably want to _not_ be allowed is the one that includes the goals (don't want to try driving through the goal post). So we should either add one of the motion constraints listed in #2700 (e.g. `AVOID_FIELD_BOUNDARY_ZONE`), or add a TODO here for it to be added as a part of that ticket.

**Review by nimazareian — CHANGES_REQUESTED** [review]
LGTM! Just had one request

File: src/software/ai/motion_constraint/motion_constraint_visitor.cpp (hunk 1 above)
Comment (itsarune) [reply src/software/ai/motion_constraint/motion_constraint_visitor.cpp:95]: done

**Review by nimazareian — APPROVED** [review]
LGTM!

**Review by PhilD71 — APPROVED** [review]
lgtm - good change, although we probably want to implement that issue to add the goal constraint soon

**Pull request merged**